osbs/build/plugins_configuration.py

```diff
@@ -0,0 +1,583 @@
+            'scratch': self.user_params.scratch.value,
+            'target': self.user_params.koji_target.value,
+            'user': self.user_params.user.value,
+            'yum_repourls': self.user_params.yum_repourls.value,
+            'arrangement_version': self.user_params.arrangement_version.value,
+            'koji_parent_build': self.user_params.koji_parent_build.value,
+            'isolated': self.user_params.isolated.value,
+        }
+
+        if self.user_params.flatpak.value:
+            build_kwargs['flatpak'] = True
+
+        self.template_set_arg(phase, plugin, 'platforms', self.user_params.platforms.value)
+        self.template_set_arg(phase, plugin, 'build_kwargs', build_kwargs)
+
+        # Parameters to be used as Configuration overrides for each worker
+        config_kwargs = {
+            'flatpak_base_image': self.user_params.flatpak_base_image.value,
+        }
+
+        # Remove empty values, and always convert to string for better interaction
+        # with Configuration class and JSON encoding
+        config_kwargs = dict((k, str(v)) for k, v in config_kwargs.items() if v is not None)
+
+        if not self.user_params.build_imagestream.value:
+            config_kwargs['build_image'] = self.user_params.build_image.value
+
+        self.template_set_arg(phase, plugin, 'config_kwargs', config_kwargs)
+
+    def render_resolve_composes(self):
+        phase = 'prebuild_plugins'
+        plugin = 'resolve_composes'
+
+        if not self.template_has_plugin_conf(phase, plugin):
+            return
+
+        if self.user_params.yum_repourls.value:
+            logger.info('removing %s from request as yum_repourls is user_paramsified', plugin)
+            self.remove_plugin(phase, plugin)
+            return
+
+        self.set_plugin_arg_valid(phase, plugin, 'compose_ids',
+                                  self.user_params.compose_ids.value)
+
+        self.set_plugin_arg_valid(phase, plugin, 'signing_intent',
+                                  self.user_params.signing_intent.value)
+
+    def render_resolve_module_compose(self):
+        phase = 'prebuild_plugins'
+        plugin = 'resolve_module_compose'
+
+        if self.template_has_plugin_conf(phase, plugin):
+            if not self.user_params.flatpak.value:
+                self.remove_plugin(phase, plugin)
+                return
+
+            self.set_plugin_arg_valid(phase, plugin, 'compose_ids',
+                                      self.user_params.compose_ids.value)
+
+    def render_squash(self):
+        phase = 'prepublish_plugins'
+        plugin = 'squash'
+
+        if self.user_params.flatpak.value:
+            # We'll extract the filesystem anyways for a Flatpak instead of exporting
+            # the docker image directly, so squash just slows things down.
+            self.remove_plugin(phase, plugin)
+            return
+
+    def render_tag_from_config(self):
```

Comment:
FYI: there's been some changes to parent's version of this method.

Reply:
There have been?  It doesn't look substantially different.

Reply:
See https://github.com/projectatomic/osbs-client/pull/728/files